distributed/shuffle/_shuffle_extension.py

```diff
@@ -353,6 +401,15 @@ async def shuffle_inputs_done(self, shuffle_id: ShuffleId) -> None:
                 logger.critical(f"Shuffle inputs done {shuffle}")
                 await self._register_complete(shuffle)
 
+    async def shuffle_fail(self, shuffle_id: ShuffleId, message: str) -> None:
+        try:
+            shuffle = self.shuffles.pop(shuffle_id)
```

Comment:
Why pop the shuffle here? If we just kept it we wouldn't need the `Extension.erred_shuffle` dict would we?